python/cudf/cudf/core/column/column.py

```diff
@@ -321,6 +327,38 @@ def to_array(self, fillna=None):
         """
         return self.to_gpu_array(fillna=fillna).copy_to_host()
 
+    def fill(self, fill_value, begin=0, end=-1, inplace=False):
+
+        if end == -1:
+            end = self.size
+
+        assert end <= self.size
+        assert begin <= end
+
+        if begin == end or begin == self.size:
+            return self if inplace else self.copy()
+
+        fill_value = Scalar(fill_value)
+
+        if not inplace:
+            return libcudfxx.filling.fill(self, begin, end, fill_value)
+
+        fixed_width = True
+
+        if is_string_dtype(self.dtype):
+            fixed_width = False
+
+        if is_categorical_dtype(self.dtype):
+            fixed_width = False
```

Comment:
Is there any reason we can't fill categoricals inplace actually? It should be just encoding the `fill_value` and then filling the `codes` column inplace.

Reply:
Ahh, ok. I was thinking of categoricals as a dictionary, but didn't realize it had a predetermined set of discrete values.